### ISSUE TITLE
MSID: 87193 Version: 2 DOI: osf.io/uk7hr

### ISSUE THREAD

**Issue by nlisgo** [elifesciences/enhanced-preprints-import#1150]
[PLACE MANUSCRIPT AND EDITOR DETAILS HERE WHEN AVAILABLE] See step 3

[PLACE PDF URL HERE WHEN AVAILABLE] See step 7

## Step 1. Inform bioRxiv

Who can help: @QueenKraken, @nlisgo, @scottaubrey

- [x] Manuscript is in data hub index (https://data-hub-api.elifesciences.org/enhanced-preprints/docmaps/v1/by-publisher/elife/get-by-manuscript-id?manuscript_id=87193)

## Step 2. Create preview of manuscript

Who can help: @fred-atherden, @nlisgo, @scottaubrey

- [ ] Manuscript is available for preview (https://prod--epp.elifesciences.org/preview/87193/v2)
    - [ ] Trigger import for staging: https://prod--epp.elifesciences.org/import
    - [ ] Sync data folder with s3://prod-elife-epp-data/data
    - [ ] Trigger import for staging: https://staging--epp.elifesciences.org/import
    - [ ] Sync data folder with s3://staging-elife-epp-data/data
    - [ ] Pull request created to [enhanced-preprint-data](https://github.com/elifesciences/enhanced-preprints-data/pulls)
    - [ ] Post link to pull request below

Pull request: [PLACE LINK TO PULL REQUEST HERE]

Detailed instructions: https://github.com/elifesciences/enhanced-preprints-data#add-a-manuscript

## Step 3: Awaiting public reviews and QC

Who can help: Production team

- [ ] QC passed or QC not needed

Who can help: Editorial team

- [x] Public reviews available on sciety (https://sciety.org/articles/activity/10.31234/osf.io/uk7hr)
- [ ] Manuscript and editor details at the top of this issue (Supplied by Editorial team)
- [x] Reviews returned by EPP - https://prod--epp.elifesciences.org/api/reviewed-preprints/87193/v2/reviews

<details>
<summary>Example</summary>

```
"msas": "Genetics and Genomics", "Neuroscience"
"msid": "84628"
"version": "1"
"preprintDoi": "10.1101/2022.10.28.514241"
"articleType": "Reviewed Preprint"
"status": "Published from the original preprint after peer review and assessment by eLife."

"Reviewed Preprint posted": "2023-01-02"
"Sent for peer review": "2022-10-28"
"Posted to bioRxiv": "2022-11-21" (link: "Go to bioRxiv": "https://www.biorxiv.org/content/10.1101/2022.10.28.514241v1")

Editors:
Reviewing Editor
Michael B Eisen
University of California, Berkeley, United States
Senior Editor
Michael B Eisen
University of California, Berkeley, United States
```

</details>

## Step 4: Modify manuscripts.json (no PDF)

- [ ] Reviewed preprint is published to EPP (https://elifesciences.org/reviewed-preprints/87193)
    - [ ] Pull request created to [enhanced-preprint-client](https://github.com/elifesciences/enhanced-preprints-client/pulls)
    - [ ] Pull request switched to "Ready for review"
    - [ ] Post link to pull request below
- [ ] Request that a doi be registered
    - [ ] Post link to [#enhanced-preprint](https://elifesciences.slack.com/archives/C03EVJSUA77) comment below

Pull request: [PLACE LINK TO PULL REQUEST HERE]
[#enhanced-preprint](https://elifesciences.slack.com/archives/C03EVJSUA77) comment thread: [PLACE LINK TO COMMENT HERE]

<details>
<summary>Instructions to modify manuscripts.json</summary>

- Visit: https://github.com/elifesciences/enhanced-preprints-client/actions/workflows/publish-manuscript.yaml
- Click: Run workflow
- Complete the form and click "Run workflow"
- A successful run should result in a new pull request at https://github.com/elifesciences/enhanced-preprints-client/pulls
- Open the pull request and click the "Ready for review" button to trigger tests
- Once the tests pass and you are happy with the changes the PR can be merged

Example pull request: https://github.com/elifesciences/enhanced-preprints-client/pull/334/files

Once the pull request is merged in it should be available a few minutes later.

</details>

### Request that a doi

Post the following in [#enhanced-preprint](https://elifesciences.slack.com/archives/C03EVJSUA77):

```
@Fred can you register a doi for https://elifesciences.org/reviewed-preprints/87193
```

## Step 5: Awaiting search reindex

- [ ] Reviewed preprint is avaliable on journal homepage (https://elifesciences.org)

The search reindex is triggered once an hour. We need the reviewed preprint to be indexed as the search application serves the journal homepage.

<details>
<summary>Additional info</summary>

If needed, the jenkins pipeline to reindex search can be triggered sooner.

https://alfred.elifesciences.org/job/process/job/process-reindex-reviewed-preprints/

</details>

## Step 6: Published! Request PDF generation

- [ ] Post the link to the PDF url at the top of the issue
    - [ ] Post link to [#enhanced-preprint](https://elifesciences.slack.com/archives/C03EVJSUA77) comment below

[#enhanced-preprint](https://elifesciences.slack.com/archives/C03EVJSUA77) comment thread: [PLACE LINK TO COMMENT HERE]

Post the following to the [#enhanced-preprint](https://elifesciences.slack.com/archives/C03EVJSUA77) on slack:

```
@Ryan Dix-Peek please can you generate a PDF for https://elifesciences.org/reviewed-preprints/87193
```

## Step 7: Introduce PDF to data folder and git repo

- [ ] PDF is avaliable at https://github.com/elifesciences/enhanced-preprints-data/raw/master/data/87193/v2/87193-v2.pdf

Detailed instructions: https://github.com/elifesciences/enhanced-preprints-data#add-a-pdf

## Step 8: Add PDF url to manuscripts.json

- [ ] Reviewed preprint PDF is available for download (https://prod--epp.elifesciences.org/reviewed-preprints/87193)
    - [ ] Pull request created to [enhanced-preprint-client](https://github.com/elifesciences/enhanced-preprints-client/pulls)
    - [ ] Pull request switched to "Ready for review"
    - [ ] Post link to pull request below

[PLACE LINK TO PULL REQUEST HERE]

<details>
<summary>Instructions to add PDF url to manuscripts.json</summary>

- Visit: https://github.com/elifesciences/enhanced-preprints-client/actions/workflows/add-pdf-url-to-manuscript.yaml
- Click: Run workflow
- Complete the form and click "Run workflow"
- A successful run should result in a new pull request at https://github.com/elifesciences/enhanced-preprints-client/pulls
- Open the pull request and click the "Ready for review" button to trigger tests
- Once the tests pass and you are happy with the changes the PR can be merged

Example pull request: https://github.com/elifesciences/enhanced-preprints-client/pull/397/files

Once the pull request is merged in it should be available a few minutes later.

</details>

## Step 9: Done!

- [ ] Kettle is on!

**Comment by fred-atherden** [comment]
(Note to check table footer presence in the footer on EPP)

**Comment by fred-atherden** [comment]
Yeah I think that's fine. 

I saw that Exeter also captured it in `table-wrap-foot` (hence the note above). Anything capable of displaying  content in a  footer (which EPP is not able to) would duplicate the content (but again not a big deal):
```xml
<table-wrap id="tbl1" orientation="portrait" position="float">
<label>Table 1.</label>
<caption><p>Baseline Sociodemographic Characteristics of Participants.</p></caption>
<graphic xlink:href="uk7hrv4_tbl1.tif"/>
<table-wrap-foot>
<fn id="tf1-fn1"><label>a</label><p>Two participants in the iCBT arm were missing data for age, gender, country of residence and highest level of education.</p></fn>
<fn id="tf1-fn2"><label>b</label><p>Gender proportions were compared using Fisher&#x2019;s exact due to cell count &#x003C; 5 in the &#x2018;Other&#x2019; group.</p></fn>
</table-wrap-foot>
</table-wrap>
```

We did ask Exeter to provide `table` as well as images (both within `alternatives`), but it doesn't look like they've done so here. Can you ask that they do it going forward?

**Comment by JGilbert-eLife** [comment]
@fred-atherden They were present in the XML that Exeter provided. Did they mistag them in some way?

[87193_XML.zip](https://github.com/elifesciences/enhanced-preprints-import/files/12657527/87193_XML.zip)

**Comment by fred-atherden** [comment]
Ah great - sorry I'm being dumb. 

I checked the version that has been loaded to EPP (rather than the original package), and forgot that we have [an xsl in place](https://github.com/elifesciences/enhanced-preprints-biorxiv-xslt/blob/master/src/handle-content-alternatives.xsl) to remove the tables (and other machine readable content) when there's an image alternative because EPP only supports images (and by default the machine readable version is included in JSON where present).

**Comment by fred-atherden** [comment]
Adding blocked - I'm getting an error from EPP for the reviews - https://prod--epp.elifesciences.org/api/reviewed-preprints/87193/v2/reviews.

@HazalCiplak do you know why the DocMap only has steps for the first version with this one?

**Comment by HazalCiplak** [comment]
Hi @fred-atherden,

Thank you for bringing this to our attention! Yes, we do need to add them into the spreadsheet for each version because each version goes through a separate matching process. However, in this specific case, doing so wouldn't resolve the issue; it would actually lead to duplications by including all reviews for the same manuscript version.

To prevent this, we have a date filter in the code that excludes revisions for OSF preprints, which is why we don't currently see them. We have also created [a related ticket for this issue](https://github.com/elifesciences/data-hub-issues/issues/725), and we were hoping to receive an example to work on a solution.

The challenge with OSF URIs (in the Hypothesis data) is that they don't indicate different versions. For instance, in the URI "https://psyarxiv.com/uk7hr/", it remains the same for both versions. Please refer to the screenshot below. It seems that using the annotation_date is the potential solution to differentiate between versions.

<img width="1266" alt="Screenshot 2023-09-19 at 10 47 58" src="https://github.com/elifesciences/enhanced-preprints-import/assets/6259479/5e2b2432-2e3f-40b8-85be-0d4cb4141b6a">

cc: @de-code @QueenKraken

**Comment by fred-atherden** [comment]
Thanks @HazalCiplak, that makes sense. 

@acollings FYI on the above. Slightly tangential question - do you know what's being done for arXiv preprints? I see that they (unlike OSF) have easily accessible URIs that allow you to point to a specific version (e.g. https://arxiv.org/abs/2301.00759v1 and https://arxiv.org/abs/2301.00759v2 vs https://arxiv.org/abs/2301.00759). 

Sounds like it might potentially be important to ensure that we're using these versioned URIs when posting the reviews if we're not already (@HazalCiplak do say otherwise if I've misunderstood).

**Comment by acollings** [comment]
Thanks @fred-atherden I've passed that on to Sue: cheers

**Comment by HazalCiplak** [comment]
If it's possible, posting the OSF reviews with a versioned URI would be greatly helpful. Using a timestamp to increase the version could potentially create problems if reviews need to be updated for any reason. This is especially the case if reviews for the same version are added on different days or if reviews for a previous version need to be added again after a later version due to any mistakes, as this could lead to data issues. That's why I hesitated to update the code, in case we have the option to update the URIs in the Hypothesis data.

If it's not possible, we can certainly give it a try. However, it's important to consider that in the future, the scenarios I mentioned above could make it challenging to address potential issues. Detecting incorrect data is often more difficult than identifying missing data.

**Comment by fred-atherden** [comment]
Thanks @HazalCiplak, as I recall I think we briefly explored this in https://github.com/elifesciences/data-hub-issues/issues/718. 

As far as I can tell, it looks like OSF don't have working versioned URIs in a way that could be used unfortunately. In [the example in that ticket](https://github.com/elifesciences/data-hub-issues/issues/718#issuecomment-1614896063), their API response provides two `html` links for [the example case](https://api.osf.io/v2/files/6347ee8831d653211c2dc903/versions/):  https://osf.io/mgn32/files/osfstorage/6347ee8831d653211c2dc903?revision=2 and https://osf.io/mgn32/files/osfstorage/6347ee8831d653211c2dc903?revision=1, but when I follow them both, they both just resolve to version 2 at https://psyarxiv.com/mgn32/, so I don't know how helpful that is. The other `download` etc. links do work, but I suspect we can't use them for hypothesis?

It sounds like we may be stuck with a different approach for OSF (if you're happy to give it a try), and we would need to likely come up with a way to accommodate the issues you mention, although since that doesn't happen all too often maybe we have some time to consider how best to accommodate these changes?

**Comment by HazalCiplak** [comment]
Hi @fred-atherden ,
Thanks for reminding the exploration.

>It sounds like we may be stuck with a different approach for OSF (if you're happy to give it a try), and we would need to likely come up with a way to accommodate the issues you mention, although since that doesn't happen all too often maybe we have some time to consider how best to accommodate these changes?

I've implemented the code for OSF to utilize the Hypothesis created date for adding versions in DOIs. For future reference, please be mindful of any updates to OSF preprints Hypothesis data, considering the following:

1. For OSF preprints, it's important to post everything (reviews, eLife evaluation, and author response) together on the **same day**. Currently, the editorial team follows this practice, so it shouldn't be an issue.

2. In case of any issues with any of these (reviews, eLife evaluation, or author response), it's necessary to delete all of them and then add them back on the **same day**.

3. Furthermore, if reviews, eLife evaluation, or author response for a previous version need to be re-added after a later version due to any mistakes, it's critically important to delete the later version and then add them back **at least one day after that previous version was added again**.

I hope those make sense. Thank you.

FYI @acollings

**Comment by fred-atherden** [comment]
Hi @HazalCiplak - many thanks. And thanks for that info, good to know. @acollings does that sounds feasible to you?

**Comment by acollings** [comment]
Thanks @HazalCiplak and @fred-atherden. Yes, I'll need to alert the editorial team and Ed Office, but can I please check what we mean by OSF preprints? Anything with ODF in the DOI? Most notably https://psyarxiv.com/? Cheers!

**Comment by HazalCiplak** [comment]
> Thanks @HazalCiplak and @fred-atherden. Yes, I'll need to alert the editorial team and Ed Office, but can I please check what we mean by OSF preprints? Anything with ODF in the DOI? Most notably https://psyarxiv.com/? Cheers!

Thank you, @acollings, for bringing this to the team's attention. 
You're absolutely correct. When I referred to 'OSF preprints,' I was specifically addressing items with URIs start with 'https://psyarxiv.com/' (which will be without version) in the Hypothesis data.
